timelock-api/src/main/conjure/timelock-api.yml

```diff
@@ -228,27 +228,30 @@ types:
       LeaderTimes:
         fields:
           leaderTimes: map<Namespace, LeaderTime>
-      TimestampLeaseRequest:
-         fields:
-           requestId:
-             type: uuid
-             safety: safe
-           numFreshTimestamps:
-             type: integer
-             safety: safe
+      RequestId:
+        alias: uuid
+        safety: safe
       TimestampLeaseRequests:
-         alias: map<TimestampLeaseName, TimestampLeaseRequest>
+          fields:
+            requestsId: RequestId
+            numFreshTimestamps: map<TimestampLeaseName, integer>
```

Comment:
This is a "transaction-batch". It will grab a single lease that handles all the locks needed.

Reply:
Yap